akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -287,18 +293,14 @@ private[akka] class AffinityPool(
       // Returns true if it executed something, false otherwise
       def executeNext(): Boolean = {
         val c = q.poll()
-        if (c ne null) {
-          executing = true
-          try
-            c.run()
-          finally
-            executing = false
```

Comment:
@ktoso Last minute optimization to avoid having to do 2 volatile writes per execution, now does the idle tracking inside the IdleStrategy. FSMs ftw.